src/renderer/hooks/usePrivateReply.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback } from 'react'
+
+import useChat from './useChat'
+import { BackendRemote } from '../backend-com'
+import { createChatByContactId } from '../backend/chat'
+
+import type { T } from '@deltachat/jsonrpc-client'
+
+export type PrivateReply = (
+  accountId: number,
+  message: T.Message
+) => Promise<void>
+
+export default function usePrivateReply() {
+  const { selectChat } = useChat()
+
+  return useCallback<PrivateReply>(
+    async (accountId, message) => {
+      const quotedMessageId = message.id
+      const contactId = message.fromId
+      const chatId = await createChatByContactId(accountId, contactId)
+
+      // retrieve existing draft to append the quotedMessageId
+      const oldDraft = await BackendRemote.rpc.getDraft(accountId, chatId)
+
+      await BackendRemote.rpc.miscSetDraft(
+        accountId,
+        chatId,
+        oldDraft?.text || null,
+        oldDraft?.file || null,
+        quotedMessageId,
+        'Text'
```

Comment:
I wonder if this is correct, we should remember this pace in the code and take over the existing view type, at best we should make a new core/jsonrpc api for "add quoted message to existing draft".

with the composer refactoring this will definitely become incorrect as it recreates the message and so it would delete all status updates of a prepared webxdc message, it should modify the existing draft if it exists and not recreate the draft message with a new message id.

Reply:
@adzialocha I know you just copied it over, this issue is older, I'm not blaming you ;)

Reply:
Do you want to create an issue for that?

Reply:
https://github.com/deltachat/deltachat-desktop/issues/3750